doc/zephyr.doxyfile.in

```diff
@@ -279,6 +279,9 @@ ALIASES                = "rst=\verbatim embed:rst:leading-asterisk" \
                          "req{1}=\ref ZEPH_\1 \"ZEPH-\1\"" \
                          "satisfy{1}=\xrefitem satisfy \"Satisfies requirement\" \"Requirement Implementation\" \1" \
                          "verify{1}=\xrefitem verify \"Verifies requirement\" \"Requirement Verification\" \1" \
+                         "kconfig_dep{1}=\attention Available only when the follow Kconfig option is enabled: \kconfig{\1}." \
+                         "kconfig_dep{2}=\attention Available only when the follow Kconfig options are enabled: \kconfig{\1}, \kconfig{\2}." \
+                         "kconfig_dep{3}=\attention Available only when the follow Kconfig options are enabled: \kconfig{\1}, \kconfig{\2}, \kconfig{\3}." \
```

Comment:
I guess we'll need to add more if we ever have more than 3 Kconfig dependencies? 

How do we deal with `CONFIG_FOO or CONFIG_BAR`? We will likely see some of those if we start applying this tree-wide

Reply:
> I guess we'll need to add more if we ever have more than 3 Kconfig dependencies?

yup - unfortunately I don't think there is any way to make this smarter, but I am hopeful that 3 should be enough to begin with.

> How do we deal with `CONFIG_FOO or CONFIG_BAR`? We will likely see some of those if we start applying this tree-wide

maybe `kconfig_dep_all` and `kconfig_dep_any`? but then someone will probably ask about `CONFIG_FOO and CONFIG_BAR or CONFIG_BAZ` :)

Reply:
What about something like `kconfig_dep_all(CONFIG_FOO && CONFIG_BAR || CONFIG_BAZ)`? Of course that would either require a logic parser when generating the text, or just inserting "CONFIG_FOO && CONFIG_BAR || CONFIG_BAZ" verbose into the attention note.